hawkrest/__init__.py

```diff
@@ -23,11 +23,39 @@
 default_message_expiration = 60
 
 
+def default_credentials_lookup(cr_id):
+    if cr_id not in settings.HAWK_CREDENTIALS:
+        raise LookupError('No Hawk ID of {id}'.format(id=cr_id))
+    return settings.HAWK_CREDENTIALS[cr_id]
+
+
+def default_user_lookup(request, credentials):
+    return HawkAuthenticatedUser(), None
+
+
 class HawkAuthentication(BaseAuthentication):
 
-    def authenticate(self, request):
-        on_success = (HawkAuthenticatedUser(), None)
+    def _credentials_lookup(self, cr_id):
+        lookup = default_credentials_lookup
+        lookup_name = getattr(
+            settings,
+            'HAWK_CREDENTIALS_LOOKUP',
+            None)
+        if lookup_name:
```

Comment:
I hadn't put any in because you didn't have any there originally. Can do, though.